### PR TITLE
gnrc: fix link-layer/IPv6 address conversion for XBee

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Fixes for auto-address generation some issues @kYc0o had during the release tests.

### Issues/PRs references
none

### CODE REVIEW

**Comment by kYc0o** [comment]
Please squash.

File: sys/net/gnrc/netif/gnrc_netif.c [hunk 1]
```diff
@@ -770,7 +770,7 @@ int gnrc_netif_ipv6_group_idx(gnrc_netif_t *netif, const ipv6_addr_t *addr)
 }
 
 #if defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_CC110X) || \
-    defined(MODULE_NRFMIN)
+    defined(MODULE_NRFMIN) || defined(MODULE_XBEE)
```
Comment (bergzand) [inline sys/net/gnrc/netif/gnrc_netif.c]: The closing `#endif` does not get the `|| defined(MODULE_XBEE)` added  ([here](https://github.com/RIOT-OS/RIOT/pull/8490/files#diff-3eb20543397a7e6bb7064e71fac0471bR785))

**Comment by bergzand** [comment]
@kYc0o Looks good to me. From what I see here, from an netif perspective, an xbee module is almost identical to a generic ieee802.15.4 radio?

**Comment by miri64** [comment]
> From what I see here, from an netif perspective, an xbee module is almost identical to a generic ieee802.15.4 radio?

Nope, you talk to xbee with AT commands and the hardware takes a lot of the header parsing from you. So this is why it doesn't use `netdev_ieee802154` as a dependency. "Generic" IEEE 802.15.4 usually use SPI or I2C to talk to the CPU and you have a lot more control over the header and addressing.

**Comment by smlng** [comment]
please squash and provide backport

**Comment by miri64** [comment]
Squashed and amended fix for @bergzand's [comment](https://github.com/RIOT-OS/RIOT/pull/8490#discussion_r164883156).

**Comment by miri64** [comment]
Backport provided in #8495

**Pull request merged**